application/src/objects/abstract-marker.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var Phaser = require('phaser');
+var _      = require('underscore');
+
+var AbstractMarker = function(state, x, y, key, angle, length)
+{
+    var validAngles;
+
+    Phaser.Sprite.apply(this, [state.game, x, y, key]);
+
+    this.state = state;
+
+    // This will make the added width add to both sides of the sprite
+    // equally. Without it, the x and y passed in will specify the position
+    // of the very left edge of the marker
+    this.anchor.setTo(0.5, 0.5);
+
+    this.height = 32;
+    this.width  = length;
+
+    validAngles = [0, 90, 180, 270];
+    if (! _(validAngles).contains(angle)) {
+        throw new Error('Invalid marker angle. Must be one of: ' + validAngles.join(', '));
+    }
```

Comment:
The `overlap` method is very imprecise for rotated shapes, and it is the only way to accomplish collision detection for objects that aren't part of a physics system. It seems to calculate the bounds of a rotated rectangle as the smallest non-rotated rectangle that can encompass it. If I allowed other angles, the markers would activate well before reaching them. Although precision probably doesn't matter that much for markers (except perhaps for the finish line) it seems best to just not allow angles that aren't multiples of 90. I don't think the limitation will matter too much, and I don't remember MM ever respawning cars at any angle inbetween the ones I'm allowing.